serving/using-external-dns.md

```diff
@@ -0,0 +1,171 @@
+# Use ExternalDNS to automate DNS setup
+
+[ExternalDNS](https://github.com/kubernetes-incubator/external-dns) is a tool 
+that synchronizes exposed Kubernetes Services and Ingresses with DNS providers.
+
+This doc introduces how to set up ExternalDNS within Knative cluster to 
+automate the process of publishing the domain used in Knative.
+
+## Prerequisite
+
+1. A Kubernetes cluster with [Knative Serving](https://github.com/knative/docs/blob/master/install/README.md) installed.
+1. Own the domain that will be used in Knative.
+1. Follow the steps [Edit using kubectl](https://github.com/knative/docs/blob/master/serving/using-a-custom-domain.md#edit-using-kubectl) to make Knative 
+use your owned domain.
+
+## Setup steps
+
+In this document, we use the GCP as an example to show how to set up 
+ExternalDNS. You can find detailed instructions for other cloud providers in [ExternalDNS documentation](https://github.com/kubernetes-incubator/external-dns#deploying-to-a-cluster).
+
+### Choose a DNS provider
+
+Skip this step if you already have a DNS provider for your owned domain.
+
+Here is a [list](https://github.com/kubernetes-incubator/external-dns#the-latest-release-v05) of DNS providers supported by ExternalDNS.
+Choose a DNS provider from the list.
+
+### Create a DNS zone for managing DNS records
+
+Skip this step if you already have a zone for managing DNS records of your 
+custom domain.
+
+A DNS zone which will contain the managed DNS records needs to be created.
+Assume your custom domain is `external-dns-test.my-org.do`. You can use the 
+following command to create a DNS zone if you are using Google Cloud DNS.
```

Comment:
Can we split a sub section here for Google Cloud DNS specific stuff?  That way other people could contribute other provider instructions easily.

Reply:
Done